vsomeip-sys/build.rs

```diff
@@ -0,0 +1,264 @@
+        format!("-I{}", arch_specific_cpp_stdlib).as_str(),
+        format!("-I{}", vsomeip_interface_path.display()).as_str(),
+    ])
+    .build()?;
+    b.flag_if_supported("-std=c++17")
+        .flag_if_supported("-Wno-deprecated-declarations") // suppress warnings from C++
+        .flag_if_supported("-Wno-unused-function") // compiler compiling vsomeip
+        .compile("autocxx-portion");
+    println!("cargo:rerun-if-changed=src/lib.rs");
+    println!("cargo:rustc-link-lib=vsomeip3");
+    println!("cargo:rustc-link-search=native={}", vsomeip_lib_path);
+
+    let include_dir = project_root.join("src/glue"); // Update the path as necessary
+
+    // we use cxx to generate bindings for those couple of functions for which autocxx fails
+    // due to usage of std::function inside of the function body
+    cxx_build::bridge("src/cxx_bridge.rs")
+        .file("src/glue/application_registrations.cpp")
+        .file("src/glue/src/application_wrapper.cpp")
+        .include(&include_dir)
+        .include(&vsomeip_interface_path)
+        .include(&runtime_wrapper_dir)
+        .flag_if_supported("-Wno-deprecated-declarations") // suppress warnings from C++
+        .flag_if_supported("-Wno-unused-function") // compiler compiling vsomeip
+        .flag_if_supported("-std=c++17")
+        .extra_warnings(true)
+        .compile("cxx-portion");
+    println!("cargo:rerun-if-changed=src/cxx_bridge.rs");
+
+    // we rewrite the autocxx generated code to suppress the cargo warning about unused imports
+    let file_path = Path::new(&out_dir)
+        .join("autocxx-build-dir")
+        .join("rs")
+        .join("autocxx-ffi-default-gen.rs");
+    println!("cargo:warning=# file_path : {}", file_path.display());
+
+    if !file_path.exists() {
+        panic!("Unable to find autocxx generated code to rewrite");
+    }
+
+    // Run rustfmt on the file
+    let status = Command::new("rustfmt")
+        .arg(&file_path)
+        .status()
+        .expect("Failed to execute rustfmt");
+
+    if !status.success() {
+        panic!("Failed to format autocxx generated file");
+    }
+
+    // Open the input file for reading
+    let input_file = File::open(&file_path).expect("Failed to open the input file for reading");
+    let reader = BufReader::new(input_file);
+
+    // Create a temporary file for writing the modified content
+    let temp_file_path = file_path.with_extension("tmp");
+    let temp_file =
+        File::create(&temp_file_path).expect("Failed to create a temporary file for writing");
+    let mut writer = BufWriter::new(temp_file);
+
+    for line in reader.lines() {
+        let mut line = line.expect("Failed to read a line from the input file");
+        line = line.replace(
+            "pub use bindgen::root::std_chrono_duration_long_AutocxxConcrete",
+            "#[allow(unused_imports)] pub use bindgen::root::std_chrono_duration_long_AutocxxConcrete"
+        );
+        line = line.replace(
```

Comment:
Do we not want to create a patch and apply it like what Ivan was doing for the vsomeip helloworld example? Actually looking over the code again it looks like you are overriding the rust generated code is that correct? Why?

Reply:
Yes, overriding the generated Rust code in a few spots to:
1. eliminate warnings
2. add the ability to output enum fields